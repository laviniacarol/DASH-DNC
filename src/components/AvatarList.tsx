import { StyledH2, StyledSpan } from '@/components'
import { Avatar, Box } from '@mui/material'
import { pxToRem } from '@/utils'
import { AvatarListProps } from '@/types'



function AvatarList(props: AvatarListProps) {
    return (
        <>
        {
          props.listData.map((item, index) => (
            <Box 
              key={index}
              sx={{
              alignItems: 'center',
              display: 'flex',
              padding: `${pxToRem(12)} 0`,
            }}>

         <Box>
         <Avatar
          alt={item.name}
          src={item.avatar}
          sx={{
            width: pxToRem(48),
            height: pxToRem(48),
            marginRight: pxToRem(16)
          }}
         />
         </Box>
          <Box>
            <StyledH2 color='inherit'>{item.name}</StyledH2>
            <StyledSpan color='inherit'>{item.subtitle}</StyledSpan>
          </Box>
        </Box>
          ))}
        </>
    )
}

export default AvatarList