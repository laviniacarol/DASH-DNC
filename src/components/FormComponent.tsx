import { FormComponentProps } from "@/types";

function FormComponent(props: FormComponentProps) {
    const { input, button, message } = props
    return (
        <form>
            {input.map((inputProps, index) => (
              <input key={index} {...inputProps} />
            ))}
             {button.map((buttonProps, index) => (
              <button key={index} {...buttonProps} />
            ))}
            {
                message && (<div style={{color: message.type === 'error' ? 'red' : 'green'}}>
                    {message.msg}
                </div>)
            }
        </form>
    )
}

export default FormComponent;