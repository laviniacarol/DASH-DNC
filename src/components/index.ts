export { default as CustomChart} from './CustomChart'
export { BannerImage } from './BannerImage'
export { default as CustomTable } from './CustomTable'
export * from './Logo'
export { default as AvatarList } from './AvatarList'
export { default as FormComponent} from './FormComponent'
export { default as Header} from './Header'
export * from './CardComponent'
export * from './StyleButton'
export * from './StyledInput'
export * from './Typographies'
