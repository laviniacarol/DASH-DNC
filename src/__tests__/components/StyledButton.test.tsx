import { StyledButton} from '@/components'
import { render } from '@testing-library/react'
import { DefaultTheme } from 'styled-components';
import { ThemeProvider } from 'styled-components'
import { themeList } from '../../resources/themesList'

describe('StyledButton', () => {
  const renderComponent = (theme: DefaultTheme,className?: string, props = {}) =>
    render(
      <ThemeProvider theme={theme}>
        <StyledButton className={className} {...props}/>
      </ThemeProvider>
    )

  themeList.forEach(({ name, theme }) => {
    describe(`${name}`, () => {
        it('it should match the spnapshot with alert class', () => {
        const { asFragment } = renderComponent(theme, 'alert')
        expect(asFragment()).toMatchSnapshot()
      })
         it('it should match the spnapshot with primary class', () => {
        const { asFragment } = renderComponent(theme, 'primary')
        expect(asFragment()).toMatchSnapshot()
      })
        it('it should match the spnapshot with borderless alert class', () => {
        const { asFragment } = renderComponent(theme, 'borderless-alert')
        expect(asFragment()).toMatchSnapshot()
      })
          it('it should match the spnapshot with disabled class', () => {
        const { asFragment } = renderComponent(theme, 'primary', { disabled: true})
        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})