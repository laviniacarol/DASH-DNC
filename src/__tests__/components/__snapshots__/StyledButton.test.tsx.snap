// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`StyledButton Dark Theme it should match the spnapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme it should match the spnapshot with borderless alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme it should match the spnapshot with disabled class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Dark Theme it should match the spnapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme it should match the spnapshot with alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme it should match the spnapshot with borderless alert class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 borderless-alert"
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme it should match the spnapshot with disabled class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 primary"
    disabled=""
  />
</DocumentFragment>
`;

exports[`StyledButton Light Theme it should match the spnapshot with primary class 1`] = `
<DocumentFragment>
  .c0 {
  border-radius: 0.5rem;
  border: none;
  box-sizing: border-box;
  cursor: pointer;
  font-size: 1rem;
  font-weight: bold;
  height: 3.125rem;
  padding: 0 1rem;
  transition: background-color 0.3s;
  width: 100%;
}

.c0.primary {
  background-color: #0070f3;
  color: #FFF;
}

.c0.primary:hover {
  background-color: #0059c1;
}

.c0.alert {
  background-color: #E00;
  color: #FFF;
}

.c0.alert:hover {
  background-color: #C00;
}

.c0.borderless-alert {
  background-color: none;
  color: #E00;
  padding: 0;
  height: 0;
}

.c0.borderless-alert:hover {
  color: #C00;
}

.c0:disabled {
  background-color: #CCC;
  color: #666;
  cursor: not-allowed;
}

.c0:disabled:hover {
  color: #CCC;
}

<button
    class="c0 primary"
  />
</DocumentFragment>
`;
